app/models/manageiq/providers/microsoft/infra_manager/vm.rb

```diff
@@ -1,6 +1,8 @@
 class ManageIQ::Providers::Microsoft::InfraManager::Vm < ManageIQ::Providers::InfraManager::Vm
   include_concern 'ManageIQ::Providers::Microsoft::InfraManager::VmOrTemplateShared'
 
+  supports_not :migrate, :reason => _("Migrate operation is not supported.")
```

Comment:
this should be on the base class and probably in `relocation.rb` and then only support it in providers that do so